### ISSUE TITLE
[REQUEST] ARM64 Windows installer

### ISSUE THREAD

**Issue by Beyley** [workman-layout/Workman#77]
I noticed the repo only has AMD64 and x86 installers for windows

**Comment by joel0** [comment]
This is a good idea; thank you for the request.

Unfortunately, as far as I'm aware, the Microsoft Keyboard Layout Creator does not compile the layouts for ARM64, so to accomplish this, we'll need to find a different compiler.  Until that can be done, the only recommendations I have are perhaps AutoHotKey or similar tools that do remapping without installing the layout into the OS.

I hope to eventually get an ARM64 installer, but can't give an ETA.

**Comment by Beyley** [comment]
> This is a good idea; thank you for the request.
> 
> Unfortunately, as far as I'm aware, the Microsoft Keyboard Layout Creator does not compile the layouts for ARM64, so to accomplish this, we'll need to find a different compiler. Until that can be done, the only recommendations I have are perhaps AutoHotKey or similar tools that do remapping without installing the layout into the OS.
> 
> I hope to eventually get an ARM64 installer, but can't give an ETA.

The x86 installer *does* seem to work (at least using Volterra's conversion), but a native ARM installer would be good, as i dont know how the conversion works in win10 or older

**Comment by rollingmoai** [comment]
> Unfortunately, as far as I'm aware, the Microsoft Keyboard Layout Creator does not compile the layouts for ARM64, so to accomplish this, we'll need to find a different compiler

Yep, MSKLC is abandonware. There is a discussion about this on another repo:
- https://github.com/rimas-kudelis/windows-keyboard-layouts/issues/1

**Comment by stirlhoss** [comment]
I read the other relevant threads and understand the difficulties with this. Just wanted to chime in and say that the x86 installer does not seem to work on my Thinkpad x13s. 

I am going to investigate some of the workarounds and can report back if any of them work. 

It is too bad windows makes this a pain instead of just supporting the layout officially.

**Comment by fkelava** [comment]
Hello,

This can be worked around. MSKLC, deprecated or not, is a frontend for ``KbdTool``/``kbdutool.exe``, which generates source files which can be compiled with the Windows SDK/WDK like any other project. 

Levicki wrote about this [for x64 almost two decades ago](https://levicki.net/articles/2006/09/29/HOWTO_Build_keyboard_layouts_for_Windows_x64.php), but a more pragmatic solution is just [retrofitting the ``kbdus`` Windows driver sample ](https://github.com/microsoft/Windows-driver-samples/tree/main/input/layout/kbdus) with the output of ``kbdutool.exe -u -v -s "{PATH-TO-WORKMAN-KLC}"``.

I have posted exactly this at https://github.com/fkelava/wm-us. I have no ARM64 machine to test it on myself, but I know it cleanly cross-compiles with the latest (10.0.26000.1) SDK/WDK. Would this work for your purposes?

How a layout is declared once compiled is explained succinctly in the [lelegard/winkbdlayouts](https://github.com/lelegard/winkbdlayouts#declaring-a-keyboard-layout-on-windows) repo.
